health.go

```diff
@@ -69,7 +81,7 @@ func (h *health) RegisterCheck(check Check, opts ...CheckOption) error {
 
 	result := h.updateResult(check.Name(), initialResultMsg, 0, initialErr, time.Now())
 	h.checksListener.OnCheckRegistered(check.Name(), result)
-	h.scheduleCheck(h.createCheckTask(check), cfg)
+	h.scheduleCheck(h.createCheckTask(check, cfg.executionTimeout), cfg.initialDelay, cfg.executionPeriod)
```

Comment:
I'm not saying that I disagree with this change, but @eranharel asked me to pass in the config instead. IMHO this is a perfect example showing why **not** passing the whole config to every method is a good idea: only pass the options that a function needs.